core/store/src/cold_storage.rs

```diff
@@ -136,6 +144,36 @@ fn get_keys_from_store(
             key_type,
             match key_type {
                 DBKeyType::BlockHash => vec![block_hash_key.clone()],
+                DBKeyType::ShardUId => shard_layout
+                    .get_shard_uids()
+                    .iter()
+                    .map(|uid| uid.to_bytes().to_vec())
+                    .collect(),
+                DBKeyType::TrieNodeOrValueHash => {
+                    let mut keys = vec![];
+                    for shard_uid in shard_layout.get_shard_uids() {
+                        let shard_uid_key = shard_uid.to_bytes();
+
+                        debug_assert_eq!(
+                            DBCol::TrieChanges.key_type(),
+                            &[DBKeyType::BlockHash, DBKeyType::ShardUId]
+                        );
+                        let trie_changes_option: Option<TrieChanges> =
+                            store
+                                .get_ser(
+                                    DBCol::TrieChanges,
+                                    &join_two_keys(&block_hash_key, &shard_uid_key),
+                                )?;
+
+                        if let Some(trie_changes) = trie_changes_option {
+                            for op in trie_changes.insertions() {
+                                store.insert_state_to_cache_from_op(op, &shard_uid_key);
+                                keys.push(op.hash().as_bytes().to_vec());
+                            }
```

Comment:
This would kinda break get_keys_from_store only reading data from storage, but perhaps it would be better to just write all this data to cold storage directly rather than populating the in-memory cache? DBCol::State would then need to be handled with a special case in the main data-copying loop by being skipped.

Reply:
That is a good idea. It will also be usefull for StateChanges, because we also will be writing values for that column in cache during the reading of the keys. But that is a change that I will save for the future.